src/components/calcite-alert/calcite-alert.tsx

```diff
@@ -168,8 +176,14 @@ export class CalciteAlert {
     );
     const progress =
       this.active && this.dismiss ? <div class="alert-dismiss"></div> : "";
+    const role =
+      this.active && this.dismiss
+        ? "alert"
+        : this.active
+        ? "alertdialog"
+        : null;
```

Comment:
Possibly simpler:
`const role = !this.active ? null : this.dismiss ? "alert" : "alertdialog";`